src/track/macro.cpp

```diff
@@ -0,0 +1,140 @@
+#include "track/macro.h"
+
+#include <QDebug>
+
+#include "util/assert.h"
+
+const QLoggingCategory kMacroLoggingCategory("macros");
+
+// static
+QList<MacroAction> Macro::deserialize(const QByteArray& serialized) {
+    proto::Macro macroProto = proto::Macro();
+    QList<MacroAction> result;
+    VERIFY_OR_DEBUG_ASSERT(macroProto.ParseFromArray(serialized.data(), serialized.length())) {
+        qCDebug(kMacroLoggingCategory) << "Failed parsing Macro from " << serialized;
+        return result;
+    }
+    result.reserve(macroProto.actions_size());
+    for (const proto::Macro_Action& action : macroProto.actions()) {
+        result.append(MacroAction(mixxx::audio::FramePos(action.source_frame()),
+                mixxx::audio::FramePos(action.target_frame())));
+    }
+    return result;
+}
+
+// static
+QByteArray Macro::serialize(const QList<MacroAction>& actions) {
+    proto::Macro macroProto;
+    auto* actionsProto = macroProto.mutable_actions();
+    for (const MacroAction& action : actions) {
+        actionsProto->Add(action.serialize());
+    }
+    return QByteArray::fromStdString(macroProto.SerializeAsString());
+}
+
+Macro::Macro()
+        : m_actions({}),
+          m_label(QString("")),
+          m_state(StateFlag::Enabled) {
+}
+
+Macro::Macro(const QList<MacroAction>& actions, const QString& label, State state)
+        : m_actions(actions),
+          m_label(label),
+          m_state(state) {
+}
+
+bool Macro::isDirty() const {
+    return m_bDirty;
+}
+
+DbId Macro::getId() const {
+    return m_iId;
+}
+
+QString Macro::getLabel() const {
+    return m_label;
+}
+
+void Macro::setLabel(const QString& label) {
+    setDirty();
+    m_label = label;
+}
+
+bool Macro::isEnabled() const {
+    return m_state.testFlag(StateFlag::Enabled);
+}
+
+bool Macro::isLooped() const {
+    return m_state.testFlag(StateFlag::Looped);
+}
+
+const Macro::State& Macro::getState() const {
+    return m_state;
+}
+
+void Macro::setState(StateFlag flag, bool enable) {
+    if (m_state.testFlag(flag) != enable) {
+        setDirty();
+        m_state.setFlag(flag, enable);
+    }
+}
+
+bool Macro::isEmpty() const {
```

Comment:
Why didn't you inline all these tiny and trivial (= single line) functions? Useless noise.

Reply:
Because `m_actions` is private, and it should be.